src/content/docs/en/core-concepts/astro-pages.mdx

```diff
@@ -96,3 +100,75 @@ For a custom 404 error page, you can create a `404.astro` or `404.md` file in `/
 
 This will build to a `404.html` page. Most [deploy services](/en/guides/deploy/) will find and use it.
 
+## Partials
+
+<Since v="3.4.0" />
+
+:::caution
+Partials are intended to be used in conjunction with a front-end library, such as [htmx](https://htmx.org/) or [Unpoly](https://unpoly.com/). You can also use them if you are comfortable writing low-level front-end JavaScript. For this reason they are an advanced feature.
+
+Additionally, partials should not be used if the component contains scoped styles or scripts, as these elements will be stripped from the HTML output. If you need scoped styles, it is better to use regular, non-partial pages along with a frontend library that knows how to merge the contents into the head.
+:::
+
+Partials are `.astro` components located within `src/pages/` that are not intended to render as full pages.
+
+Like `.astro` components located outside of this folder, these files do not automatically include the `<! DOCTYPE html>` declaration, nor any `<head>` content such as scoped styles and scripts.
+
+However, because they are located in the special `src/pages/` directory, the generated HTML is available at a URL corresponding to its file path. This allows a rendering library (e.g. htmx, Stimulus, jQuery) to access it on the client and load sections of HTML dynamically on a page without a browser refresh or page navigation.
+
+
+Partials, when combined with a rendering library, provide an alternative to [Astro islands](/en/concepts/islands/) and [`<script>` tags](/en/guides/client-side-scripts/) for building dynamic content in Astro.
```

Comment:
```suggestion
Partials are page components located within `src/pages/` that are not intended to render as full pages.

Like components located outside of this folder, these files do not automatically include the `<! DOCTYPE html>` declaration, nor any `<head>` content such as scoped styles and scripts.

However, because they are located in the special `src/pages/` directory, the generated HTML is available at a URL corresponding to its file path. This allows a rendering library (e.g. htmx, Stimulus, jQuery) to access it on the client and load sections of HTML dynamically on a page without a browser refresh or page navigation.

Partials, when combined with a rendering library, provide an alternative to [Astro islands](/en/concepts/islands/) and [`<script>` tags](/en/guides/client-side-scripts/) for building dynamic content in Astro.
```